### DISCUSSION TITLE
Training with a small data set

### DISCUSSION THREAD

**Discussion by hemakgp** [mrdbourke/pytorch-deep-learning#991] [Q&A]
I have a custom dataset with 719 images.  I split these into 80% for training and 20 % for testing.  Batch size 10.Tensor [10,1,28,28] . 58 batches for training. 14 batches for testing.  I tried training with  linear layers and ReLU and used all combinations one by one increasing the number of hidden units, adding additional layers, changing the optimizer (SGD to Adams). Nothing seems to work. I have visualized the images and class labels; they all are correct. I am following the exact steps mentioned by  DBourke(which is excellent). I have not tried CNN yet.  Any suggestions  for training with a small dataset will help

**Comment by Arcane-WD** [comment]
Try changing the test-train split. Worked for me on a different model...

**Reply by hemakgp** [reply]
Thanks. I tried with Train 0.9 & Test 0.1 No change.  

Epoch: 22
-------

Train loss: 0.05676 | Train_accuracy:0.00 | Test loss: 3.64037, Test acc: 3.75, Epoch:22

Epoch: 23
-------

Train loss: 0.05729 | Train_accuracy:0.00 | Test loss: 3.63877, Test acc: 3.75, Epoch:23

Epoch: 24
-------

Train loss: 0.05706 | Train_accuracy:0.00 | Test loss: 3.64048, Test acc: 3.75, Epoch:24 


What more can be done to improve the Acc?

**Reply by Arcane-WD** [reply]
By increasing train data percentage, you might be causing "overfitting" try a 70-30 or 60-40 split, if accuracy is still not good then re check what features you have chosen. I too am pretty knew to this concept so this is all I can understand from your problem 👍🏻
